### PR TITLE
Add initial enrolment flow

### PR DESCRIPTION
**Pull request by mpgxvii**
- Adds initial study landing page, eligibility, registration, verification, and consent flow
- Eligibility and consent forms are on separate ts files, support needs to be added to pull these from external repo/Github
- Add initial oauth consent module that would allow the user to consent to access to a resource server which will then allow an access token to be requested from hydra
- This also allows claims to be mapped from the kratos identity (which has the MP information) to the hydra token

Solves: 
- https://github.com/RADAR-base/hydra-kratos/issues/23, https://github.com/RADAR-base/hydra-kratos/issues/24, https://github.com/RADAR-base/hydra-kratos/issues/25, https://github.com/RADAR-base/hydra-kratos/issues/29, https://github.com/RADAR-base/hydra-kratos/issues/28, https://github.com/RADAR-base/hydra-kratos/issues/31, https://github.com/RADAR-base/hydra-kratos/issues/13

### CODE REVIEW

File: data/eligibility-questionnaire.ts [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+      section_header: "",
+      field_type: "text",
+      field_label: "Eligible",
+      select_choices_or_calculations: "",
+      field_note: "",
+      text_validation_type_or_show_slider_number: "",
+      text_validation_min: "",
+      text_validation_max: "",
+      identifier: "",
+      branching_logic: "",
```
Comment (mpgxvii) [inline data/eligibility-questionnaire.ts:77]: Branching logic to check for eligibiliy will be added here. Example:
```
    "branching_logic": "[has_fitbit] = '1' && [city] = 'London'"
```

File: pages/profile.tsx [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+              return Promise.reject(err)
+            }),
+        )
+    )
+  }
+
+  return (
+    <>
+      <Head>
+        <title>Profile Page</title>
```
Comment (yatharthranjan) [inline pages/profile.tsx:136]: I think we need something generic here that can parse as many Identifiable questionnaires as necessary

**Review by yatharthranjan — COMMENTED** [review]
Thanks, excellent work!

Please take a look at some comments below. Most, if not all, of these can be done in separate issues/PRs later.


As discussed before, it would be good to abstract most of the input types so they can be reused across various pages.

**Review by this-Aditya — COMMENTED** [review]
Hi Pauline,

Overall, this PR looks excellent. I’ve left some comments focused on minor enhancements. Please take a look at these comments and make any adjustments you deem appropriate.

**Comment by yatharthranjan** [comment]
@mpgxvii Can you also run Prettier to fix the formatting issues?

**Review by Bdegraaf1234 — APPROVED** [review]
Great work Pauline,

I think the others reviewed this better than I could have, LGTM

File: pages/eligibility.tsx [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+// Renders the eligibility page
+const Eligibility: NextPage = () => {
+  const IS_ELIGIBLE = "yes"
+  const router = useRouter()
+  const [eligibility, setEligibility] = useState(null)
+  const questions: any[] = eligibilityQuestions
+
+  const checkEligibility = async (values: any) => {
+    // Eligibility check
+    return values.is_eligible == IS_ELIGIBLE
```
[the remark this answers is not in the document]
Comment (mpgxvii) [reply pages/eligibility.tsx:27]: Issue added here: https://github.com/RADAR-base/hydra-kratos/issues/48

File: pages/eligibility.tsx [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+    if (!router.isReady) {
+      return
+    }
+  })
+
+  const onSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault()
+    const formData = new FormData(event.currentTarget)
+    const formValues = Object.fromEntries(formData.entries())
+    const eligible = await checkEligibility(formValues)
```
[the remark this answers is not in the document]
Comment (mpgxvii) [reply pages/eligibility.tsx:42]: Thanks for the suggestions, made the changes! 👍

File: pages/registration.tsx [hunk 5]
```diff
@@ -19,17 +19,28 @@ const Registration: NextPage = () => {
   // The "flow" represents a registration process and contains
   // information about the form we need to render (e.g. username + password)
   const [flow, setFlow] = useState<RegistrationFlow>()
+  const [eligibility, setEligibility] = useState<any>()
+  const [projectId, setProjectId] = useState<any>()
 
   // Get ?flow=... from the URL
   const { flow: flowId, return_to: returnTo } = router.query
```
[the remark this answers is not in the document]
Comment (mpgxvii) [reply pages/registration.tsx:27]: Made the changes 👍

File: pages/study.tsx [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+      </Head>
+      <MarginCard>
+        <CardTitle>{projectId} Research Study</CardTitle>
+        <img src="image.png" />
+        <StudyInfo questions={studyInfo} />
+        <TextCenterButton
+          className=""
+          disabled=""
+          data-testid=""
+          href="/eligibility"
```
[the remark this answers is not in the document]
Comment (mpgxvii) [reply pages/study.tsx]: Thanks, updated this. 👍

**Pull request merged**